### DISCUSSION TITLE
How can I add new collection from code?

### DISCUSSION THREAD

**Discussion by lonelyButterflyAtWork** [directus/directus#8233] [Archive]
Hi, 
I want to add a new table and a collection to my database, but I need to do it using code and something like migrations because I have three environments with their own databases.

I tried to add a migration file and then use npx directus database migrate:up command but the message I got was 'nothing to migrate'. 

I cannot manage tables in directus in the production application so my second question is: Is there a way to create a collection from a table that was added using sql in the db panel?

**Comment by joselcvarela** [comment]
Your only option is to stick with migrations.
Have you followed this: https://docs.directus.io/guides/migrations/#custom-migrations ?

Did you already ran that migration?
You can check the `directus_migrations` table.

In order to help you further, we need you to share the migration relative path, migration code and the result of `select * from directus_migrations`.

**Comment by rijkvanzanten** [comment]
Custom migrations is one way to do it, alternatively, you can rely on the just-released Schema Migrations capabilities, that allows you to snapshot the schema of the database as a flat file you can then use to provision the schema of the other projects:

https://docs.directus.io/reference/command-line-interface/#migrate-schema-to-a-different-environment
